### PR TITLE
Makefilecleanup

### PR DESCRIPTION
**Pull request by jedwards4b**
Fix clean build for clm shared versions, clean up Makefile logic a little

case.clean_build was not working properly for clm

Test suite: ran several I and B cases along with yellowstone prealpha drv suite
Test baseline: cesm1_5_alpha06b
Test namelist changes: none
Test status: bit for bit

Fixes: 

Code review:

### CODE REVIEW

File: cime_config/cesm/machines/Makefile [hunk 1]
```diff
@@ -776,8 +770,8 @@ cleanrof:
 	cd $(EXEROOT)/rof/obj ; $(RM) -f *.o *.mod
 
 cleanlnd:
-	$(RM) -f $(LIBROOT)/liblnd.a
-	cd $(EXEROOT)/lnd/obj ; $(RM) -f *.o *.mod
+	$(RM) -f $(LNDLIBDIR)/$(LNDLIB)
```
Comment (billsacks) [inline cime_config/cesm/machines/Makefile:773]: For cleanlnd: The new version looks more correct than the old version. However, we have seen in the past that changes to the clean_build behavior can cause unexpected test failures. What testing have you done to ensure that the new clean_build behavior will work correctly in a test suite? I would like to see a demonstration that this works correctly for a test suite that includes a number of clm45/clm50 tests, including some that do two builds. The yellowstone intel aux_clm45 test suite would be a good example.

File: scripts/Tools/xmlchange [hunk 2]
```diff
@@ -143,7 +143,7 @@ if ( ! defined($settinglist) ) {
    $idlist{$opts{'id'}} = $opts{'val'};
 } else {
    foreach my $varval ( split( /,/, $settinglist ) ) {
-      if ( $varval =~ /^([a-zA-Z0-9_]+)=([^,=]+)$/ ) {
+       if ( $varval =~ /^(\w+)=(.*+)$/ ){
```
Comment (billsacks) [inline scripts/Tools/xmlchange:146]: This change looks unrelated to the rest of the pull request, right? Ideally this would be in a separate pull request. But at the very least can you document the purpose of this change and the testing you have done to ensure that it is correct?

**Comment by billsacks** [comment]
The Makefile cleanup here looks like some good cleanup and fixes - thanks! In addition to my line comments, I would like to see more specific documentation of what testing you have done. Specifically: Have you tested the build for a clm40, clm45 and clm50 case, with verification that each of these leads the lnd library to be created in the right place (clm45 & clm50 in the shared lib location, clm4 not)?

File: cime_config/cesm/machines/Makefile (hunk 1 above)
Comment (jedwards4b) [inline cime_config/cesm/machines/Makefile:773]: I have tested the yellowstone prealpha intel test suite which does include several I cases.   Is there a suite that tests all three flavors of clm?   Is there a baseline I can compare to?

Comment (billsacks) [inline cime_config/cesm/machines/Makefile:773]: I'm running the yellowstone-intel aux_clm45 test suite on this: I realized it was more efficient for me to just do this myself. I'll add a note later (probably this evening) with the results.

Comment (billsacks) [inline cime_config/cesm/machines/Makefile:773]: Okay, yellowstone-intel aux_clm45 test suite looks good (run from clm4_5_8_r171 with cime updated to this branch)
